File: Medium/1561. Maximum Number of Coins You Can Get/1561. Maximum Number of Coins You Can Get.rs
impl Solution {
    pub fn max_coins(mut piles: Vec<i32>) -> i32 {
        piles.sort_unstable(); // Use unstable sort to improve performance

        let mut ans = 0;
        let mut ind = piles.len() - 2;
        let iterats = piles.len() / 3;

        for _ in 0..iterats {
            ans += piles[ind];
            ind -= 2;
        }

        ans
    }
}

File: Medium/1561. Maximum Number of Coins You Can Get/1561. Maximum Number of Coins You Can Get.ts
function maxCoins(piles: number[]): number {
    piles.sort((a, b) => a - b);

    let res = 0;

    for (let i = Math.floor(piles.length / 3); i < piles.length; i += 2) {
        res += piles[i];
    }

    return res;
}
